src/app.controller.ts

```diff
@@ -1,14 +1,50 @@
-import { Controller, Get } from '@nestjs/common';
+import {
+  Controller,
+  Get,
+  HttpException,
+  HttpStatus,
+  Param,
+  Res,
+} from '@nestjs/common';
 import { AppService } from './app.service';
 import { Public } from './auth/auth.decorator';
+import { LinksService } from './links/links.service';
+import { UrlService } from './url/url.service';
+import { Response } from 'express';
 
 @Controller()
 export class AppController {
-  constructor(private readonly appService: AppService) {}
+  constructor(
+    private readonly appService: AppService,
+    private readonly linksService: LinksService,
+    private readonly urlService: UrlService,
+  ) {}
 
   @Public()
   @Get()
   getHello(): string {
     return this.appService.getHello();
   }
+
+  @Public()
+  @Get(':shortUrl') //  GET /links/:id
+  async getLinkByShortUrl(
+    @Param('shortUrl') shortUrl: string,
+    @Res() res: Response,
+  ) {
+    const originalUrl = await this.linksService.findOrginalUrl(shortUrl);
+
+    if (!originalUrl) {
+      throw new HttpException('Nie znaleziono linku', HttpStatus.FORBIDDEN);
```

Comment:
**suggestion**: Use a more descriptive error message for better clarity to the client.